packages/core/src/App/Components/Elements/LiveChat/use-livechat.ts

```diff
@@ -0,0 +1,135 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useHistory } from 'react-router';
+import { liveChatInitialization } from './live-chat';
+import Cookies from 'js-cookie';
+import { deriv_urls } from '@deriv/shared';
+
+// Todo: Should move this declaration into a .d.ts file.
+declare global {
+    interface Window {
+        LiveChatWidget: {
+            on: (key: string, callback: VoidFunction) => void;
+            get: (key: string) => any;
+            call: (key: string, value?: any) => void;
+        };
+        LC_API: {
+            on_chat_ended: VoidFunction;
+        };
+    }
+}
+
+const useLiveChat = (has_cookie_account = false) => {
```

Comment:
@mitra-deriv This hook body is a bit large and I think we can break it into smaller hooks or utility functions later as improvement, Maybe you can leave a todo here for the next person if you agree 🙇🏻

Reply:
> @mitra-deriv This hook body is a bit large and I think we can break it into smaller hooks or utility functions later as improvement, Maybe you can leave a todo here for the next person if you agree 🙇🏻

Sure Farzin, Totally agree, I will add a todo and try to improve it thanks 🙏 🌺